pyseas/maps/bivariate.py

```diff
@@ -0,0 +1,239 @@
+            scale = 1 / aspect_ratio
+
+        if height is None:
+            height = width * scale
+        if width is None:
+            width = height / scale
+
+    return width, height
+
+
+def add_bivariate_colorbox(bvcmap, xnorm=None, ynorm=None, *, ax=None, fig=None, xlabel='', ylabel='',
+                xformat=None, yformat=None, loc='below right', width=None, height=None, aspect_ratio=None,
+                bg_color=None, fontsize=8, pad=0.05):
+    """Add colorbar to a PySeas raster
+
+    Parameters
+    ----------
+    bvcmap : BivariateCmap
+    xnorm, ynorm : matplotlib normalizer
+    ax : matplotlib Axes, optional
+    fig : matplotlib Figure, optional
+    xlabel, ylabel : str, optional
+    xformat, yformat : str, optional
+    loc : str or tuple of float, optional
+        Location name or ocation of the colorbox in axis coordinates.  
+        Currently supported names are 'below right', 'lower right', 'upper right',
+        and 'above right'.
+    width, height : float or None, optional
+        Size of colorbar in Axes coordinates. If only one is specified and the last
+        projection was one of the standard one, then an appropriate aspect ratio will
+        be used. Otherwise the same width and height will be used.
+    aspect_ratio : float or None, optional
+    bg_color : maplotlib color, optional
+    fontsize : int, optional # TODO: stylize
+    pad : float, optional
+    
+    Returns
+    -------
+    Axes
+    """
+    if ax is None:
+        ax = plt.gca()
+    if xnorm is None:
+        xnorm = Normalize(vmin=0, vmax=1, clip=True)
+    if ynorm is None:
+        ynorm = Normalize(vmin=0, vmax=1, clip=True)
+    width, height = _setup_width_and_height(width, height, aspect_ratio)
+    if isinstance(loc, str):
+        loc = _loc_finder(loc, width, height, pad=pad)
+    wloc, hloc = loc
+
+    cb_ax = ax.inset_axes([wloc, hloc, width, height], transform=ax.transAxes)
+
+    x, y = np.meshgrid(np.linspace(xnorm.vmin, xnorm.vmax, 1000), 
+                       np.linspace(ynorm.vmin, ynorm.vmax, 1000))
+
+    im = cb_ax.imshow(bvcmap(xnorm(x), ynorm(y)), 
+        extent=(xnorm.vmin, xnorm.vmax, ynorm.vmin, ynorm.vmax), origin='lower', aspect='auto')
+    bg_color = bg_color or plt.rcParams.get('pyseas.ocean.color', props.dark.ocean.color)
+
+    cb_ax.set_facecolor(bg_color)
+
+    cb_ax.set_xlabel(xlabel, fontsize=fontsize)
+    cb_ax.set_ylabel(ylabel, fontsize=fontsize)
+
+    if _is_log(bvcmap.log_x, xnorm):
+        cb_ax.set_xscale('log')
+    if _is_log(bvcmap.log_y, ynorm):
+        cb_ax.set_yscale('log')
+
+    if xformat is not None:
+        cb_ax.xaxis.set_major_formatter(xformat)
+    if yformat is not None:
+        cb_ax.yaxis.set_major_formatter(yformat)
+
+
+    plt.sca(ax)     
+    return cb_ax
+
+
+
+def add_bivariate_raster(raster1, raster2, bvcmap=_default_cmap, norm1=None, norm2=None, *,
```

Comment:
Suggestion: rename variables ending in `1` and `2` to be more description like `raster_x`, `norm_x`, etc. Or perhaps following some of your other naming, it should be `xraster` and `xnorm`